operator/validator/observability.go

```diff
@@ -0,0 +1,76 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/ssvlabs/ssv/observability"
+)
+
+const (
+	observabilityName      = "github.com/ssvlabs/ssv/operator/validator"
+	observabilityNamespace = "ssv.validator"
+)
+
+type validatorStatus string
+
+const (
+	statusError        validatorStatus = "error"
+	statusNotFound     validatorStatus = "not_found"
+	statusReady        validatorStatus = "ready"
+	statusSlashed      validatorStatus = "slashed"
+	statusExiting      validatorStatus = "exiting"
+	statusNotActivated validatorStatus = "not_activated"
+	statusPending      validatorStatus = "pending"
+	statusNoIndex      validatorStatus = "no_index"
+	statusRemoved      validatorStatus = "removed"
+	statusUnknown      validatorStatus = "unknown"
+)
+
+var (
+	meter = otel.Meter(observabilityName)
+
+	validatorStatusGauge = observability.NewMetric(
+		meter.Int64Gauge(
+			metricName("status"),
+			metric.WithDescription("validator status")))
+)
+
+func metricName(name string) string {
+	return fmt.Sprintf("%s.%s", observabilityNamespace, name)
+}
+
+func validatorStatusAddrAttribute(value validatorStatus) attribute.KeyValue {
+	attrName := fmt.Sprintf("%s.status", observabilityNamespace)
+	return attribute.String(attrName, string(value))
+}
+
+func recordValidatorStatus(ctx context.Context, status validatorStatus) {
+	resetExecutionClientStatusGauge(ctx)
+
+	validatorStatusGauge.Record(ctx, 1,
+		metric.WithAttributes(validatorStatusAddrAttribute(status)),
+	)
+}
+
+func resetExecutionClientStatusGauge(ctx context.Context) {
+	for _, status := range []validatorStatus{
+		statusError,
+		statusNotFound,
+		statusReady,
+		statusSlashed,
+		statusExiting,
+		statusNotActivated,
+		statusPending,
+		statusNoIndex,
+		statusRemoved,
+		statusUnknown} {
+		validatorStatusGauge.Record(ctx, 0,
+			metric.WithAttributes(validatorStatusAddrAttribute(status)),
+		)
```

Comment:
In_ OTel, since we don't have some equivalent for `Reset()`, do you think recording `0` is the most adequate to signal that we don't have a value for a given metric, vs something such as `-1`? 

Since all of these statuses theoretically cover all possible states for a validator, I'm wondering if it's worth tracking and actually using the gauges to increase and decrease rather than resetting and counting every time, is it possible to verify that the final behavior is the same from a visualization perspective? e.g. you can see validators transitioning from one state to another.